lib/src/backend/mod.rs

```diff
@@ -0,0 +1,57 @@
+use anyhow::{anyhow, Result};
+use clap::ValueEnum;
+
+use ostree::glib;
+use ostree_ext::container::OstreeImageReference;
+use ostree_ext::keyfileext::KeyFileExt;
+use ostree_ext::ostree;
+use ostree_ext::sysroot::SysrootLock;
+
+use crate::spec::ImageStatus;
+
+mod ostree_container;
+
+pub(crate) trait Backend {
+    fn backend(&self) -> Result<Box<dyn BackendImpl>>;
+}
+
+pub(crate) trait BackendImpl {
+    fn imagestatus(
+        &self,
+        sysroot: &SysrootLock,
+        deployment: &ostree::Deployment,
+        image: OstreeImageReference,
+    ) -> Result<(Option<ImageStatus>, Option<ImageStatus>)>;
+}
+
+impl Backend for crate::spec::Backend {
```

Comment:
The duplication of `Backend` here feels odd...WDYT about having something like:

```
pub(crate) fn load(ty: crate::spec::Backend) -> Result<Box<dyn BackendImpl>> {
  match ty {
          crate::spec::Backend::OstreeContainer => {
                Ok(Box::new(ostree_container::OstreeContainerBackend))
            }
  }
}
```

or so?

IOW it's only `Deployment` that has an `impl Backend`?

Reply:
> the only thing we're going to have to guide which backend to use will be whatever is provided via the spec::Backend

Right...though we could "resolve" that early and pass down the backend as an argument into lower layers.

Here's one way I'm thinking about it, we could have an abstraction over `ostree::Sysroot` - like `bootc::Storage` that has an `Option<bootc:ContainerStorage>` or something, and if that's not-none then we use it.